File: source/src/main/java/com/kingmang/lazurite/parser/optimization/VariableInfo.java
package com.kingmang.lazurite.parser.optimization;

import com.kingmang.lazurite.runtime.Value;

public final class VariableInfo {
    public Value value;
    int modifications;

    @Override
    public String toString() {
        return (value == null ? "?" : value) + " (" + modifications + " mods)";
    }
}

File: src/main/java/com/kingmang/lazurite/optimization/VariableInfo.kt
package com.kingmang.lazurite.optimization

import com.kingmang.lazurite.runtime.values.LzrValue


class VariableInfo {
    @JvmField
    var value: LzrValue? = null
    @JvmField
    var modifications: Int = 0

    override fun toString(): String {
        return (if (value == null) "?" else value).toString() + " (" + modifications + " mods)"
    }
}